### ISSUE TITLE
error response when using class-validator is incorrect

### ISSUE THREAD

**Issue by sincraianul** [nestjs/nest#1622]
## I'm submitting a...

<pre><code>
[x] Regression 
[ ] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

If you follow the example in the [validation section of the docs](https://docs.nestjs.com/techniques/validation) and use class-validator to annotate class members, the response from the server consists only of the errors array:

```json
[
  {
    "target": {},
    "property": "email",
    "children": [],
    "constraints": {
      "isEmail": "email must be an email"
    }
  }
]
```

## Expected behavior

Server response should be:

```json
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": [
    {
      "target": {},
      "property": "email",
      "children": [],
      "constraints": {
        "isEmail": "email must be an email"
      }
    }
  ]
}
```

## Minimal reproduction of the problem with instructions

Just follow the instructions in the docs.

## Environment

<pre><code>
Nest version: 5.7.3

</code></pre>

## Suggested fix
This issue occurs due to the changes brought by #1344. I can PR the behaviour and add an additional detection for `isObject` to exclude objects that are arrays.

**Comment by kamilmysliwiec** [comment]
>  I can PR the behaviour and add an additional detection for isObject to exclude objects that are arrays.

Sounds perfect :)

**Comment by sincraianul** [comment]
fixed by #1623

**Comment by kamilmysliwiec** [comment]
Published as 5.7.4
